### PR TITLE
Brings experimental cloner in line

### LINKED ISSUE #42099: Experimental cloner bugged out
Issue reported from  Round ID: 99327 (/tg/Station Bagil [ENGLISH] [US-EAST] [100% FREE LAG])

Reporting client version: 512

reproduction: 

- find prototype cloner
- take it (pod and scanner) back to the station WITHOUT the console
- build it with a normal console
- set it to autoprocess
- get in 
- wait for the clone to come out
- it's named "e1de1d4ab630963b99d4e"

### PR DESCRIPTION
**Pull request by 4dplanner**
:cl: 
fix: experimental cloner is compatible with ordinary cloning computer, but deletes records after cloning.
/:cl:

fixes #42099

### CODE REVIEW

**Comment by AlexTheSysop42** [comment]
Don't make it delete records or it'll burn all of them if it's hooked up to the normal one as a extra pod

**Comment by Swiftkor** [comment]
phew thought you were gonna delet my favorite thing about space

**Comment by dr-bee** [comment]
> 
> 
> phew thought you were gonna delet my favorite thing about space

at the moment it continually makes catatonic bodies if hooked up to a regular cloning console.

**Comment by 4dplanner** [comment]
> at the moment it continually makes catatonic bodies if hooked up to a regular cloning console.

weird thing to say on the PR to fix that

File: code/__DEFINES/machines.dm [hunk 1]
```diff
@@ -102,3 +102,7 @@
 #define MACHINE_NOT_ELECTRIFIED 0
 #define MACHINE_ELECTRIFIED_PERMANENT -1
 #define MACHINE_DEFAULT_ELECTRIFY_TIME 30
+
+//cloning defines. These are flags.
+#define CLONING_SUCCESS 1<<0
+#define CLONING_DELETE_RECORD 1<<1
```
Comment (Cyberboss) [inline code/__DEFINES/machines.dm]:
```suggestion
 #define CLONING_DELETE_RECORD (1<<1)
```

File: code/__DEFINES/machines.dm [hunk 2]
```diff
@@ -102,3 +102,7 @@
 #define MACHINE_NOT_ELECTRIFIED 0
 #define MACHINE_ELECTRIFIED_PERMANENT -1
 #define MACHINE_DEFAULT_ELECTRIFY_TIME 30
+
+//cloning defines. These are flags.
+#define CLONING_SUCCESS 1<<0
```
Comment (Cyberboss) [inline code/__DEFINES/machines.dm]:
```suggestion
 #define CLONING_SUCCESS (1<<0)
```

File: code/game/machinery/cloning.dm [hunk 3]
```diff
@@ -139,32 +139,32 @@
...
 		if(G.suiciding) // The ghost came from a body that is suiciding.
-			return FALSE
+			return 0
 	if(clonemind.damnation_type) //Can't clone the damned.
 		INVOKE_ASYNC(src, .proc/horrifyingsound)
 		mess = TRUE
 		icon_state = "pod_g"
 		update_icon()
-		return FALSE
+		return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/cloning.dm [hunk 4]
```diff
@@ -139,32 +139,32 @@
...
+			return 0
 		if(clonemind.current.suiciding) // Mind is associated with a body that is suiciding.
-			return FALSE
+			return 0
 	if(!clonemind.active)
 		// get_ghost() will fail if they're unable to reenter their body
 		var/mob/dead/observer/G = clonemind.get_ghost()
 		if(!G)
-			return FALSE
+			return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 			return NONE
```

File: code/game/machinery/cloning.dm [hunk 5]
```diff
@@ -139,32 +139,32 @@
...
 		if(G.suiciding) // The ghost came from a body that is suiciding.
-			return FALSE
+			return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 			return NONE
```

File: code/game/machinery/cloning.dm [hunk 6]
```diff
@@ -139,32 +139,32 @@
...
 	if(clonemind.last_death != last_death) //The soul has advanced, the record has not.
-		return FALSE
+		return 0
 	if(!QDELETED(clonemind.current))
 		if(clonemind.current.stat != DEAD)	//mind is associated with a non-dead body
-			return FALSE
+			return 0
 		if(clonemind.current.suiciding) // Mind is associated with a body that is suiciding.
-			return FALSE
+			return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 			return NONE
```

File: code/game/machinery/cloning.dm [hunk 7]
```diff
@@ -139,32 +139,32 @@
...
 	if(!istype(clonemind))	//not a mind
-		return FALSE
+		return 0
 	if(clonemind.last_death != last_death) //The soul has advanced, the record has not.
-		return FALSE
+		return 0
 	if(!QDELETED(clonemind.current))
 		if(clonemind.current.stat != DEAD)	//mind is associated with a non-dead body
-			return FALSE
+			return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 			return NONE
```

File: code/game/machinery/cloning.dm [hunk 8]
```diff
@@ -139,32 +139,32 @@
...
 	if(mess || attempting)
-		return FALSE
+		return 0
 	clonemind = locate(mindref) in SSticker.minds
 	if(!istype(clonemind))	//not a mind
-		return FALSE
+		return 0
 	if(clonemind.last_death != last_death) //The soul has advanced, the record has not.
-		return FALSE
+		return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/cloning.dm [hunk 9]
```diff
@@ -139,32 +139,32 @@
...
 	if(panel_open)
-		return FALSE
+		return 0
 	if(mess || attempting)
-		return FALSE
+		return 0
 	clonemind = locate(mindref) in SSticker.minds
 	if(!istype(clonemind))	//not a mind
-		return FALSE
+		return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/cloning.dm [hunk 10]
```diff
@@ -139,32 +139,32 @@
 //Start growing a human clone in the pod!
 /obj/machinery/clonepod/proc/growclone(clonename, ui, mutation_index, mindref, last_death, datum/species/mrace, list/features, factions, list/quirks, datum/bank_account/insurance)
 	if(panel_open)
-		return FALSE
+		return 0
 	if(mess || attempting)
-		return FALSE
+		return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/cloning.dm [hunk 11]
```diff
@@ -139,32 +139,32 @@
 //Start growing a human clone in the pod!
 /obj/machinery/clonepod/proc/growclone(clonename, ui, mutation_index, mindref, last_death, datum/species/mrace, list/features, factions, list/quirks, datum/bank_account/insurance)
 	if(panel_open)
-		return FALSE
+		return 0
```
Comment (Cyberboss) [inline code/game/machinery/cloning.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/exp_cloner.dm [hunk 12]
```diff
@@ -9,7 +9,7 @@
 	internal_radio = FALSE
 
 //Start growing a human clone in the pod!
-/obj/machinery/clonepod/experimental/growclone(ckey, clonename, ui, se, datum/species/mrace, list/features, factions)
+/obj/machinery/clonepod/experimental/growclone(clonename, ui, mutation_index, mindref, last_death, datum/species/mrace, list/features, factions, list/quirks, datum/bank_account/insurance)
 	if(panel_open)
 		return FALSE
```
Comment (Cyberboss) [inline code/game/machinery/exp_cloner.dm]:
```suggestion
 		return NONE
```

File: code/game/machinery/exp_cloner.dm [hunk 13]
```diff
@@ -9,7 +9,7 @@
...
 	if(mess || attempting)
```
Comment (Cyberboss) [inline code/game/machinery/exp_cloner.dm]: Below this line too

**Review by Cyberboss — CHANGES_REQUESTED** [review]

**Comment by AlexTheSysop42** [comment]
The bodies aren't always catatonic afaik, it's that autocloning makes a lot of bodies and quickly depletes the ghost pool

File: code/__DEFINES/machines.dm (hunk 1 above)
Comment (ExcessiveUseOfCobblestone) [reply code/__DEFINES/machines.dm]: the space...

File: code/__DEFINES/machines.dm (hunk 2 above)
Comment (ExcessiveUseOfCobblestone) [reply code/__DEFINES/machines.dm]: AHHHHHHHHHHHHHH

**Pull request merged**